dbms/src/Flash/FlashService.cpp

```diff
@@ -274,6 +276,89 @@ grpc::Status FlashService::BatchCoprocessor(grpc::ServerContext * grpc_context,
     return ret;
 }
 
+grpc::Status FlashService::CoprocessorStream(grpc::ServerContext * grpc_context,
+                                             const coprocessor::Request * request,
+                                             grpc::ServerWriter<coprocessor::Response> * writer)
+{
+    CPUAffinityManager::getInstance().bindSelfGrpcThread();
+    bool is_remote_read = getClientMetaVarWithDefault(grpc_context, "is_remote_read", "") == "true";
+    auto log_level = is_remote_read ? Poco::Message::PRIO_INFORMATION : Poco::Message::PRIO_DEBUG;
+    LOG_IMPL(log, log_level, "Handling coprocessor stream request, is_remote_read: {}, start ts: {}, region info: {}, region epoch: {}", is_remote_read, request->start_ts(), request->context().region_id(), request->context().region_epoch().DebugString());
+
+    auto check_result = checkGrpcContext(grpc_context);
+    if (!check_result.ok())
+        return check_result;
+
+    GET_METRIC(tiflash_coprocessor_request_count, type_cop_stream).Increment();
+    GET_METRIC(tiflash_coprocessor_handling_request_count, type_cop_stream).Increment();
+    if (is_remote_read)
+    {
+        GET_METRIC(tiflash_coprocessor_request_count, type_remote_read).Increment();
+        GET_METRIC(tiflash_coprocessor_handling_request_count, type_remote_read).Increment();
+    }
+    Stopwatch watch;
+    SCOPE_EXIT({
+        GET_METRIC(tiflash_coprocessor_handling_request_count, type_cop_stream).Decrement();
+        GET_METRIC(tiflash_coprocessor_request_duration_seconds, type_cop_stream).Observe(watch.elapsedSeconds());
+        // TODO: update the value of metric tiflash_coprocessor_response_bytes.
+        if (is_remote_read)
+            GET_METRIC(tiflash_coprocessor_handling_request_count, type_remote_read).Decrement();
+    });
+
+    context->setMockStorage(mock_storage);
+
+    const auto & settings = context->getSettingsRef();
+    auto handle_limit = settings.cop_pool_handle_limit != 0 ? settings.cop_pool_handle_limit.get() : 10 * cop_pool->size();
+    auto max_queued_duration_ms = std::min(settings.cop_pool_max_queued_seconds, 20) * 1000;
```

Comment:
Maybe need to use a large value for cop stream? What is the timeout threshold in client-c for cop stream?